caddyhttp/staticfiles/fileserver.go

```diff
@@ -121,13 +125,41 @@ func (fs FileServer) serveFile(w http.ResponseWriter, r *http.Request, name stri
 		return http.StatusNotFound, nil
 	}
 
+	filename := d.Name()
+
+	for _, encoding := range staticEncodingPriority {
+		if strings.Contains(r.Header.Get("Accept-Encoding"), encoding) {
+			gf, err := fs.Root.Open(location + staticEncoding[encoding])
+
+			if err == nil {
+				stat, err := gf.Stat()
+
+				if err == nil {
+					// Close previous file - release fd
+					f.Close()
```

Comment:
@wendigo @mholt You could also replace [this line](https://github.com/mholt/caddy/blob/cffeb22f522dd9488cf21a2ec423ae079bb76dd2/caddyhttp/staticfiles/fileserver.go#L69) with the following:

```go
defer func() { _ = f.Close() }()
```

That way no file handle is closed twice...
...as long as you remember to always close `f` before assigning it a new value, but considering how "dangerous" Go is anyways already I'd consider such a requirement quite normal.